src/main/scala/com/cognite/sdk/scala/v1/Client.scala

```diff
@@ -280,6 +263,24 @@ object GenericClient {
         )
       }
   }
+
+  def parseResponse[T, R](uri: Uri, mapResult: T => R)(
+      implicit decoder: Decoder[T]
+  ): ResponseAs[R, Any] =
+    asJsonEither[CdpApiError, T].mapWithMetadata((response, metadata) =>
+      response match {
+        case Left(DeserializationException(_, error)) =>
+          throw SdkException(
+            s"Failed to parse response, reason: ${error.getMessage}",
+            Some(uri),
+            metadata.header("x-request-id"),
+            Some(metadata.code.code)
+          )
+        case Left(HttpError(cdpApiError, _)) =>
+          throw cdpApiError.asException(uri"$uri", metadata.header("x-request-id"))
+        case Right(value) => mapResult(value)
+      }
```

Comment:
Did this change, or just move? If it's unintended, let's move it back.

Reply:
It's intended, because this was a private method of the class `RequestSession`, I moved it inside object `GenericClient` so we can import and use it in OAuth2.scala